gpu/test/base/gpu_executor.cu

```diff
@@ -142,6 +146,37 @@ TEST_F(GpuExecutor, CopiesDataFromGpu)
     gpu->free(orig);
 }
 
+/* Properly checks if it works only when multiple GPUs exist */
+TEST_F(GpuExecutor, PreservesStreamSettings)
+{
+    auto last_device = gko::GpuExecutor::get_num_devices() - 1;
+    cudaSetDevice(last_device);
+    auto orig = gpu->alloc<int>(2);
+    int current_stream;
+    cudaGetDevice(&current_stream);
```

Comment:
It seems like this is mixing up "device" and "stream" - you call the identifier `current_stream`, but then set it to device id...